### ISSUE TITLE
SSL Failure on 2.1.4

### ISSUE THREAD

**Issue by Gramps** [godotengine/godot#10795]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Windows 10, Ubuntu Linux 16.04, Godot Engine 2.1.4 Stable

**Issue description:**
SSL fails to connect.  Also gives a really weirdly spaced error message in console, noted below.

```
Cause: unable to get local issuer certificate
CERT STR: /C=GB/ST=Greater Manchester/L=Salford/O=COMODO CA Limited/CN=COMODO RSA Certification Authority
VALID: 0
CONNECTION RESULT: -1
8028:error:14090086:SSL routines:ssl3_get_server_certificate:certificate verify failed:thirdparty\openssl\ssl\s3_clnt.c:1269:
E R R O R :   S t r e a m P e e r O p e n S S L : : _ p r i n t _ e r r o r :   S o m e   I / O   e r r o r   o c c u r r e d .   T h e   O p e n S S L   e r r o r   q u e u e   m a y   c o n t a i n   m o r e   i n f o r m a t i o n   o n   t h e   e r r o r .
       A t :   m o d u l e s \ o p e n s s l \ s t r e a m _ p e e r _ o p e n s s l . c p p : 4 1 5
 [ 2 0 1 7 - 0 8 - 3 0   2 0 : 0 2 : 1 6 . 0 2 9 ]   D E B U G :   P O S T   r e q u e s t   s e n d i n g   d a t a
 E R R O R :   S t r e a m P e e r O p e n S S L : : _ p r i n t _ e r r o r :   A   f a i l u r e   i n   t h e   S S L   l i b r a r y   o c c u r r e d ,   u s u a l l y   a   p r o t o c o l   e r r o r .
       A t :   m o d u l e s \ o p e n s s l \ s t r e a m _ p e e r _ o p e n s s l . c p p : 4 1 8
```

**Steps to reproduce:**
Try to connect via HTTP.connect with SSL enabled and certificate added in settings.  Above response is given in console and the function fails to connect.

Have tested it in 2.1.3 and it works fine, as normal, but fails in 2.1.4.  Think it's broken.

**Comment by akien-mga** [comment]
Is this a certificate you can share to try to reproduce the issue?

**Comment by Gramps** [comment]
Yep.  I tested it with the ca-certificate.crt from Linux.

Further testing out of curiosity.  My game connects to my server which is HTTPS and redirects any non-HTTPS connection to the secure version.  In 2.1.3 with SSL not set up, this works fine.  In 2.1.4 it fails with a 302 error and responds with an error page instead of the expected JSON response.

When adding in the certificate attached and changing my HTTP.connect from
`80, false, true`
to
`443, true, true`
2.1.3 works perfectly with the certificate and connects as expected.  However, when testing it today while typing this up, it seems 2.1.4 is now working on both Windows and Linux with this certificate and HTTP.connect set up.   Not sure why it failed or how I got that error I listed initially.

Not using the certificate with an HTTPS server but normal HTTP.connect(80, false, true) still fails on 2.1.4 but not 2.1.3 which might be more of a bug on 2.1.3 than 2.1.4.

Seems it is fine now, sorry for the waste of time!

[ca-certificates.zip](https://github.com/godotengine/godot/files/1268292/ca-certificates.zip)
